cluster-autoscaler/cloudprovider/digitalocean/digitalocean_cloud_provider.go

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package digitalocean
+
+import (
+	"io"
+	"os"
+
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+	"k8s.io/autoscaler/cluster-autoscaler/config"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/errors"
+	"k8s.io/klog"
+)
+
+var _ cloudprovider.CloudProvider = (*digitaloceanCloudProvider)(nil)
```

Comment:
What is the purpose of this line?

Reply:
This verifies at compile time that `digitaloceanCloudProvider` implements the interface `cloudprovider.CloudProvider`.

Reply:
Let's us catch issues if someone modifies the interface. So you'll see immediately that the digitalocean cloud provider doesn't satisfy the interface.

Reply:
Makes sense. Thanks for clarification.